### PR TITLE
Added sphinx sphinx-autodoc-typehints

### PR DESCRIPTION
**Pull request by schreiberx**

### CODE REVIEW

**Comment by schreiberx** [comment]
@arporter Ready for merge.

File: src/psyclone/psyir/nodes/call.py
Comment (schreiberx) [inline src/psyclone/psyir/nodes/call.py:1]: I had to fix this docstring to pass the CI tests.

File: src/psyclone/psyir/nodes/call.py [hunk 1]
```diff
@@ -245,18 +245,16 @@ def insert_named_arg(self, name, arg, index):
...
 
-    def replace_named_arg(self, existing_name, arg):
+    def replace_named_arg(self, existing_name: str, arg: DataNode):
         '''Replace one named argument node with another node keeping the
         same name.
 
-           :param str existing_name: the argument name.
-           :param arg: the argument expression.
-           :type arg: :py:class:`psyclone.psyir.nodes.DataNode`
+        :param existing_name: the argument name.
```
Comment (schreiberx) [inline src/psyclone/psyir/nodes/call.py:252]: @arporter : I had to fix this docstring to pass the CI tests.

**Review by arporter — APPROVED** [review]
Thanks Martin; nice self-contained PR :-)
Docs all build fine. New dependence is installed by `pip install .[doc]`.
Will proceed to merge.

**Pull request merged**